server/client.go

```diff
@@ -1010,8 +1010,18 @@ func (c *client) traceMsg(msg []byte) {
 	if !c.trace {
 		return
 	}
-	// FIXME(dlc), allow limits to printable payload.
-	c.Tracef("<<- MSG_PAYLOAD: [%q]", msg[:len(msg)-LEN_CR_LF])
+
+	origLength := len(msg) - LEN_CR_LF
+	var msgToWrite []byte
+	if origLength > c.srv.opts.MaxTracedMessagePrintableLen {
```

Comment:
opts should be grabbed under a lock, but only if we know we are tracing.

Reply:
Will do.